### ISSUE TITLE
Bug: Creating sequential unbalanced benchmark (iNaturalist)

### ISSUE THREAD

**Issue by Mattdl** [ContinualAI/avalanche#625]
I'm trying to implement the iNaturalist2018 dataset and benchmark (#298 ).
Each supercategory is an experience, for which I use the neat interface of passing a dataset sequence.

    def SplitInaturalist(...):
      if super_categories is None:
          super_categories = [
              'Amphibia', 'Animalia', 'Arachnida', 'Aves', 'Fungi',
              'Insecta', 'Mammalia', 'Mollusca', 'Plantae', 'Reptilia']
    
      list_train_dataset, list_test_dataset = [], []
      # for every incremental experience
      for idx, supcat in enumerate(super_categories):
          # Isolate each super category as an experience
          train_set, test_set = _get_inaturalist_dataset(
              root, [supcat], download=download and idx == 0)
  
          list_train_dataset.append(train_set)
          list_test_dataset.append(test_set)
  
      if return_task_id:
          return nc_benchmark(
              train_dataset=list_train_dataset,
              test_dataset=list_test_dataset,
              n_experiences=len(super_categories),
              task_labels=True,
              # per_exp_classes=per_exp_classes,
              seed=seed,
              class_ids_from_zero_in_each_exp=True,
              train_transform=train_transform,
              eval_transform=eval_transform,
              one_dataset_per_exp=True
          )
      else:
          return nc_benchmark(
              train_dataset=list_train_dataset,
              test_dataset=list_test_dataset,
              n_experiences=len(super_categories),
              task_labels=False,
              # per_exp_classes=per_exp_classes,
              seed=seed,
              train_transform=train_transform,
              eval_transform=eval_transform,
              one_dataset_per_exp=True
          )

However, there seems to be some problems when calling the nc_benchmark.
In [this line of code](https://github.com/ContinualAI/avalanche/blob/258f2f34ea733f35aafe392160b33f4eeeb4f98c/avalanche/benchmarks/generators/benchmark_generators.py#L186) the `seq_train_dataset.targets=[341, 424, 332, 416, 3563, 3749, 3577, 3214, 3231, 2263]`.
This is completely wrong given that 
`per_exp_classes = {0: 321, 1: 178, 2: 262, 3: 2031, 4: 1258, 5: 284, 6: 144, 7: 2917, 8: 234, 9: 114}` and `len(fixed_class_order)=7743`, which are both correct for the dataset. 

I think the main problem is in this merging of the dataset in the sequence (this gives errors later on in the nc_scenario, [here is the link to the assertion error](https://github.com/ContinualAI/avalanche/blob/258f2f34ea733f35aafe392160b33f4eeeb4f98c/avalanche/benchmarks/scenarios/new_classes/nc_scenario.py#L180). In this assertion, the `classes_order_original_ids` only have len 10, whereas `fixed_class_order` len is 7743 (the right length).

A reason why the merging to the sequential dataset may fail, is that the targets per datasets don't start from 0, but are e.g. in the range of 1000 to 2500 per dataset to be merged.

Can someone help out here? Thanks!

**Comment by lrzpellegrini** [comment]
Hi Matt, if I understood correctly each supercategory has its own set of classes whose ids are in a certain range. If this is the case it's probably better to create a single instance of the dataset containing all classes and then use the "fixed class order" + "number of classes per experience" parameters of `nc_benchmark` to define how experiences are created. Would this be compatible with what you're trying to achieve?

**Comment by Mattdl** [comment]
Hi @lrzpellegrini, yes exactly! It works now, thanks a lot! (:

**Comment by lrzpellegrini** [comment]
Hi @Mattdl, can you confirm that this is fixed?

**Comment by Mattdl** [comment]
Yes this should be fixed with the PR #636 , thanks!
